### PR TITLE
Update `FileLock` constructor to accept `pathlib.Path`

### PR DESCRIPTION
**Pull request by f-koehler**
This is a small fix for people who employ `pathlib.Path` for their path handling. It allows a `Path` object to be passed to the constructor of `FileLock`. It is then converted to a normal `str` when storing it inside the object leaving the rest of the library unaffected.

### CODE REVIEW

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Test and changelog? 😊

**Comment by f-koehler** [comment]
> Test and changelog? blush

I added a test using `pytest.mark.parametrize`. What are you referring to with "changelog" am I supposed to update some changelog file. Sorry if I am being stupid but I could not find any instructions in the repo/doc. Only the `docs/changelog.rst` file which I did not want to touch since you seem to be the only author of that.

So please let me know what's missing.

**Comment by gaborbernat** [comment]
> Only the `docs/changelog.rst` file which I did not want to touch since you seem to be the only author of that.

That's the one. Ignore that in past only I've edited it.

**Review by FFY00 — CHANGES_REQUESTED** [review]
Can we please make it `os.PathLike` instead? It covers way more types and the implementation is mostly the same, you just replace `str()` with `os.fspath()`.

**Comment by f-koehler** [comment]
@FFY00 and @gaborbernat, I will get to work on this next week.

**Comment by FFY00** [comment]
Sounds good, no worries :blush:

File: src/filelock/_api.py [hunk 1]
```diff
@@ -44,7 +45,7 @@ def __init__(self, lock_file: str, timeout: float = -1) -> None:
          A timeout of 0 means, that there is exactly one attempt to acquire the file lock.
         """
         # The path to the lock file.
-        self._lock_file: str = lock_file
+        self._lock_file: str = os.fspath(lock_file)  # noqa: SC200
```
Comment (gaborbernat) [inline src/filelock/_api.py]: Please remove this noqa and fix it by adding to to the whitelist.txt.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

**Review by FFY00 — APPROVED** [review]
This looks alright to me, thanks!

**Review by gaborbernat — APPROVED** [review]

**Comment by gaborbernat** [comment]
https://pypi.org/project/filelock/3.3.2/ is now out

**Pull request merged**